scripts/west_commands/build.py

```diff
@@ -119,6 +119,10 @@ def do_add_parser(self, parser_adder):
         group = parser.add_argument_group('cmake and build tool')
         group.add_argument('-c', '--cmake', action='store_true',
                            help='force a cmake run')
+        group.add_argument('--cmake-opt', action='append',
+                           help='''same as '-- cmake_opt', but it prevents
+                           using '--' (e.g. for usage in alias commands);
+                           may be given multiple times.''')
```

Comment:
Don't use both `cmake_opt` and `cmake_opts`, it is confusing. Set a default instead and extend in `_parse_remainder`

```suggestion
        group.add_argument('--cmake-opt', action='append',
                           dest="cmake_opts", default=[],
                           help='''same as '-- cmake_opt', but it prevents
                           using '--' (e.g. for usage in alias commands);
                           may be given multiple times.''')
```

(There's a debug print for `cmake_opts`)